pom.xml

```diff
@@ -1048,7 +1062,8 @@
             <dependency>
                 <groupId>cglib</groupId>
                 <artifactId>cglib</artifactId>
-                <version>2.2</version>
+                <version>2.2.2</version>
```

Comment:
@lukaszlenart With https://github.com/apache/struts/commit/518854a08565c894a18168db7777136c9f3b8f99 we got the update to asm:7 which seems to be compatible with Java 11. 

cglib:3.2.9, according to https://github.com/cglib/cglib/releases/tag/RELEASE_3_2_9, seems to try the same and therefore may be compatible with Java 11.

@JCgH4164838Gh792C124B5 do you see any chance to use the latest version of cglib by the aim of Java 11 support?

Reply:
Hello @sepe81 .

I noted that the main pom.xml has two cglib entries (one for cglib-nodep with 2.1_3 - which matches what JMock 1.2.0 indicates as a requirement) and another separate cglib entry (was 2.2, updated to 2.2.2 with this PR).

Interestingly no cglib-x.x.jar is actually packaged and the various "DEPENDENCIES" files in the jars produced by the build only list the 2.1_3 dependency.  I'm still a relative Maven newbie, but is it possible that the non-"nodep" cglib entry is a redundant entry (leftover from a past compile-time dependency) ?

Perhaps the cglib-nodep entry dependency-scope could be set to "test", since it appears to only be needed for jmock-cglib during the test regression series.  That might help with JDK 11 runtime compatibility (if not build compatibility).

Given the above, it seems that migrating to JMock 2.x (current appears to be 2.9.0) might allow upgrading cglib to the latest version (or possibly eliminate the dependency as they mention switching to ByteBuddy for JMock 2.9 - although their release comment mentions "JDK 11 still failing).

Migrating the unit tests to JMock 2.x in the Struts 2.6.x series might be an idea for the future (and should help with a JDK 11 migration goal).

Reply:
I don't think we use CGLIB anywhere in production code, it was used sometime ago in the Convention Plugin (as far I recall)

Reply:
Thanks for your investigation. In another step I will try to remove the explicit dependency cglib-nodep. If this works and the tests are fine, I'll create a new PR.

Reply:
see https://github.com/apache/struts/pull/268